packages/compass-aggregations/src/modules/pipeline-builder/pipeline-builder.ts

```diff
@@ -0,0 +1,137 @@
+import type { DataService } from 'mongodb-data-service';
+import type * as t from '@babel/types';
+import parseEJSON from 'ejson-shell-parser';
+
+import { PipelinePreviewManager } from './pipeline-preview-manager';
+import type { PreviewOptions } from './pipeline-preview-manager';
+import { PipelineParser, Stage } from './pipeline-parser';
+
+export const DEFAULT_PIPELINE = `[\n{}\n]`;
+
+export class PipelineBuilder {
+  source: string;
+  node: t.ArrayExpression | null = null;
+  stages: Stage[] = [];
+  syntaxError: SyntaxError[] = [];
+  previewManager: PipelinePreviewManager;
```

Comment:
This should not be public, if you want to only pass pipeline builder to the thunk arg instead of keeping two for now, until we do COMPASS-6167, I would at least add a todo here so that we don't forget to clean it up

Reply:
Yes. It definitely should be private. I'll add a todo in pipeline.ts where its being currently used.